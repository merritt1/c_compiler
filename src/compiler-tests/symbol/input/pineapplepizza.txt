char b;
int a(char f) {
first: b;
goto pineapple;
}
short p(int asdf) {
second: b++;
goto pizza;
}
