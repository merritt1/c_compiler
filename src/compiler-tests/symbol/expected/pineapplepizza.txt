


***ERROR: Reference to undefined label 'pineapple' -- searched in Table 2***





***ERROR: Reference to undefined label 'pizza' -- searched in Table 4***


=============== PARSE TREE ===============
signed char (b);
signed int (a(signed char (f))) {
first:
    b;
goto pineapple;
}
signed short int (p(signed int (asdf))) {
second:
    (b++);
goto pizza;
}
================= SYMBOLS ================
Symbol Table ID: 0
  b -> signed char
  a -> function(signed int, [signed char])
  p -> function(signed short int, [signed int])

Symbol Table ID: 1
  f -> signed char

Symbol Table ID: 2
  first -> label

Symbol Table ID: 3
  asdf -> signed int

Symbol Table ID: 4
  second -> label

